multiqc/modules/damageprofiler/damageprofiler.py

```diff
@@ -0,0 +1,238 @@
+        self.lgdist_fw_data   =   self.ignore_samples(self.lgdist_fw_data)
+        self.lgdist_rv_data      =   self.ignore_samples(self.lgdist_rv_data)
+
+
+        # Write parsed data to a file
+        #self.write_data_file(self.threepGtoAfreq_data, 'multiqc_damageprofiler_3pGtoAfreq')
+        #self.write_data_file(self.fivepCtoTfreq_data, 'multiqc_damageprofiler_5pCtoTfreq')
+        #self.write_data_file(self.lgdist_fw_data, 'multiqc_damageprofiler_lgdist_fw')
+        #self.write_data_file(self.lgdist_rv_data, 'multiqc_damageprofiler_lgdist_rv')
+
+        # Basic Stats Table, use generic function to add data to general table
+        self.dmgprof_misinc_stats(self.threepGtoAfreq_data, '3 Prime', 'G -> A')
+        self.dmgprof_misinc_stats(self.fivepCtoTfreq_data, '5 Prime', 'C -> T')
+
+        # Add plots
+        if len(self.threepGtoAfreq_data) > 0:
+            self.add_section ( 
+                name = '3P Misincorporation Plot',
+                plot = self.threeprime_plot()
+            )
+        if len(self.fivepCtoTfreq_data) > 0:
+            self.add_section (
+                name = '5P Misincorporation Plot',
+                plot = self.fiveprime_plot()
+            )
+        if len(self.lgdist_fw_data) > 0 and len(self.lgdist_rv_data) > 0:
+            self.add_section (
+                name = 'Forward read length distribution',
+                plot = self.lgdistplot(self.lgdist_fw_data, 'Forward')
+            )
+            self.add_section (
+                name = 'Reverse read length distribution',
+                plot = self.lgdistplot(self.lgdist_rv_data, 'Reverse')
+            )
+
+
+    #Parse our nice little JSON file
+    def parseJSON(self, f):
+        """ Parse the JSON output from DamageProfiler and save the summary statistics """
+        try:
+            parsed_json = json.load(f['f'])
+        except Exception as e:
+            print(e)
+            log.warn("Could not parse DamageProfiler JSON: '{}'".format(f['fn']))
+            return None
+        
+        #Get sample name from JSON first
+        s_name = parsed_json['metadata']['sample_name']
+        self.add_data_source(f, s_name)
+
+        
+        #Add 3' G to A data 
+        self.threepGtoAfreq_data[s_name] = parsed_json['dmg_3p']
+
+        #Add 5' C to T data
+        self.fivepCtoTfreq_data[s_name] = parsed_json['dmg_5p']
+
+        #Add lendist forward 
+        self.lgdist_fw_data[s_name] = parsed_json['lendist_fw']
+
+        #Add lendist reverse
+        self.lgdist_rv_data[s_name] = parsed_json['lendist_rv']
+
+    
+    #### Tables from here on 
+
+    def dmgprof_misinc_stats(self, dict_to_plot, readend, substitution):
+        """ Take the parsed stats from the DamageProfiler and add it to the
+        basic stats table at the top of the report """
+
+        headers = OrderedDict()
+        headers['1'] = {
+            'title': readend + ' ' + substitution + ' 1st base ',
```

Comment:
Instead of concatenating strings like this, it's better to use `format()`:
```suggestion
            'title': '{} {} 1st base'.format(readend, substitution),
```

Reply:
Done :-)